bindings/rust/wasmedge-sys/src/config.rs

```diff
@@ -0,0 +1,152 @@
+use super::wasmedge;
+pub struct Config {
+    pub(crate) ctx: *mut wasmedge::WasmEdge_ConfigureContext,
+    _private: (),
+}
+
+impl Drop for Config {
+    fn drop(&mut self) {
+        unsafe { wasmedge::WasmEdge_ConfigureDelete(self.ctx) };
+    }
+}
+
+impl Config {
+    pub fn new() -> Self {
+        Self::default()
+    }
+}
+
+impl Default for Config {
+    fn default() -> Self {
+        let ctx = unsafe { wasmedge::WasmEdge_ConfigureCreate() };
+        if ctx.is_null() {
+            panic!("failed to create WasmEdge configuration");
+        }
+        Self { ctx,  _private: ()}
+    }
+}
+
+/// Code Generate for `WasmEdge_Proposal` :
+/// 
+/// ```rust
+/// impl Config {
+///     pub fn enable_bulkmemoryoperations(self, enable: bool) -> Self {
+///         let prop = wasmedge::WasmEdge_Proposal_BulkMemoryOperations;
+///         if enable {
+///             unsafe { wasmedge::WasmEdge_ConfigureAddProposal(self.ctx, prop) };
+///         } else {
+///             unsafe { wasmedge::WasmEdge_ConfigureRemoveProposal(self.ctx, prop) };
+///         }
+///         self
+///     }
+///     pub fn has_bulkmemoryoperations(self) -> bool {
+///         let prop = wasmedge::WasmEdge_Proposal_BulkMemoryOperations;
+///         unsafe { wasmedge::WasmEdge_ConfigureHasProposal(self.ctx, prop) }
+///     }
+///     // ...
+/// }
+/// ```
+macro_rules! impl_proposal_config {
+    ($( $proposal:ident ),+ $(,)+) => {
+        impl Config {
+            paste::paste! {
+                $(
+                    pub fn [<enable_$proposal:lower:snake>](self, enable: bool) -> Self {
+                        let prop = wasmedge::[<WasmEdge_Proposal_$proposal>];
+                        if enable {
+                            unsafe { wasmedge::WasmEdge_ConfigureAddProposal(self.ctx, prop) };
+                        } else {
+                            unsafe { wasmedge::WasmEdge_ConfigureRemoveProposal(self.ctx, prop) };
+                        }
+                        self
+                    }
+
+                    pub fn [<has_$proposal:lower:snake>](self) -> bool {
+                        let prop = wasmedge::[<WasmEdge_Proposal_$proposal>];
+                        unsafe { wasmedge::WasmEdge_ConfigureHasProposal(self.ctx, prop) }
+                    }
+                )+
+            }
+        }
+    }
+}
+
+impl_proposal_config! {
+    BulkMemoryOperations,
+    ReferenceTypes,
+    SIMD,
+    TailCall,
+    Annotations,
+    Memory64,
+    Threads,
+    ExceptionHandling,
+    FunctionReferences,
+}
+
+impl Config {
+
+    // For Vm
+    pub fn reg_wasi(self) -> Self{
+        unsafe{wasmedge::WasmEdge_ConfigureAddHostRegistration(self.ctx, wasmedge::WasmEdge_HostRegistration_Wasi)};
+        self
+    }
+
+    // For AOT compiler
+
+    /// Set the optimization level of AOT compiler.
+    pub fn opt_level(self, opt_level: OptLevel) -> Self {
```

Comment:
```suggestion
    pub fn set_opt_level(self, opt_level: OptLevel) -> Self {
```
`opt_level` is the idiomatic name of a getter

Reply:
This is a reuse of code from version 0.1, not yet polished in detail, and I am not currently using clippy. formal pr will take this into account.